shpc/main/modules/base.py

```diff
@@ -67,29 +67,34 @@ def modulefile(self):
     def templatefile(self):
         return "%s.%s" % (self.container.templatefile, self.module_extension)
 
-    def uninstall(self, name, view=None, force=False):
+    def view_uninstall(self, view, name, force=False):
         """
-        Given a unique resource identifier, uninstall a module. If a view
-        name is provided, assume we only want to uninstall from the view
+        Uninstall a module from a view.
         """
-        # The name can either be a folder or an install directory
-        name = self.add_namespace(name)
+        module = self.new_module(name)
 
-        # folder paths do not have :
-        name = name.replace(":", os.sep)
-        module_dir = os.path.join(self.settings.module_base, name)
-        container_dir = self.container.container_dir(name)
+        # Ask before deleting anything!
+        if not force:
+            msg = name + "?"
+            if not utils.confirm_uninstall(msg, force):
+                return
```

Comment:
The following one makes sense to have the msg variable, but this one can be shortened to just:
```suggestion
            if not utils.confirm_uninstall(name + "?", force):
                return
```